ledger/ledger-api-client/src/test/suite/scala/com/digitalasset/ledger/client/services/commands/CommandTrackerFlowTest.scala

```diff
@@ -581,7 +582,29 @@ class CommandTrackerFlowTest
       }
     }
 
+    "successful completions arrive for the same command submitted twice with different submission IDs" should {
+
+      "output two successful responses" in {
+        val Handle(submissions, results, _, completionStreamMock) =
+          runCommandTrackingFlow(allSubmissionsSuccessful)
+
+        results.request(2)
+
+        submissions.sendNext(newSubmission("submissionId", commandId))
+        submissions.sendNext(newSubmission("anotherSubmissionId", commandId))
+
+        completionStreamMock.send(successfulCompletion("submissionId", commandId))
+        completionStreamMock.send(successfulCompletion("anotherSubmissionId", commandId))
```

Comment:
What about the case where we get a rejection and a successful completion?

Reply:
Not sure if it's worth exploring more here, as there are tests for failures in the stream.

Reply:
i.e. failures are not being handled differently than success responses

Reply:
OK. You know the implementation better and you're the one who has to be happy with the test coverage in the end.